includes/Modules/Analytics_4.php

```diff
@@ -1173,39 +1174,36 @@ function ( $metric_def ) {
 	 * Parses the orderby value of the data request into an array of AnalyticsData OrderBy object instances.
 	 *
 	 * @since 1.93.0
+	 * @since n.e.x.t Updated to provide support for ordering by dimensions.
 	 *
 	 * @param array|null $orderby Data request orderby value.
 	 * @return Google_Service_AnalyticsData_OrderBy[] An array of AnalyticsData OrderBy objects.
 	 */
 	protected function parse_reporting_orderby( $orderby ) {
-		if ( empty( $orderby ) || ! is_array( $orderby ) ) {
+		if ( empty( $orderby ) || ! is_array( $orderby ) || ! wp_is_numeric_array( $orderby ) ) {
 			return array();
 		}
 
 		$results = array_map(
 			function ( $order_def ) {
-				$order_def = array_merge(
-					array(
-						'fieldName' => '',
-						'sortOrder' => '',
-					),
-					(array) $order_def
-				);
-
-				if ( empty( $order_def['fieldName'] ) || empty( $order_def['sortOrder'] ) ) {
+				$order_by = new Google_Service_AnalyticsData_OrderBy();
+				$order_by->setDesc( isset( $order_def['desc'] ) ? $order_def['desc'] : false );
```

Comment:
Ensure the value is a boolean:

```suggestion
				$order_by->setDesc( ! empty( $order_def['desc'] ) );
```